cluster/etcd_service_discovery.go

```diff
@@ -374,10 +384,11 @@ func parseServer(value []byte) (*Server, error) {
 }
 
 func (sd *etcdServiceDiscovery) printServers() {
-	sd.serverMapByType.Range(func(k, v interface{}) bool {
-		logger.Log.Debugf("type: %s, servers: %s", k, v)
-		return true
-	})
+	sd.mapByTypeLock.RLock()
```

Comment:
Same here. Why not use `defer`?

Reply:
Defers are only useful when there multiple flows, in simple cases like this calling the function directly is a little more efficient